src/viur/core/db/transport.py

```diff
@@ -83,10 +99,23 @@ def delete(keys: t.Union[Entity, t.List[Entity], Key, t.List[Key]]):
     """
 
     _write_to_access_log(keys)
-    if not isinstance(keys, (set, list, tuple)):
-        return __client__.delete(keys)
-
-    return __client__.delete_multi(keys)
+    keys_data = []
+    single_request = isinstance(keys, (Key, Entity))
+    if isinstance(keys, Entity):
+        keys_data = keys.key
+    elif isinstance(keys, Key):
+        keys_data = keys
+    elif not single_request:
+        for key in keys:
+            if isinstance(key, Key):
+                keys_data.append(key)
+            elif isinstance(key, Entity):
+                keys_data.append(key.key)
+    cache.delete(keys_data)
+    if single_request:
+        return __client__.delete(keys_data)
+
+    return __client__.delete_multi(keys_data)
```

Comment:
Same here. Assume keys is always an iterable.

```suggestion
    keys = (k.key if isinstance(k, Entity) else k for k in utils.ensure_iterable(keys))
    if not keys:
        return
    
    cache.delete(keys)
    if len(keys) == 1:
        return __client__.delete(keys)

    return __client__.delete_multi(keys)
```

Reply:
@phorward I use get and not get_multi because it's faster.